### ISSUE TITLE
Enable WinRM automatically

### ISSUE THREAD

**Issue by mdoyle-cvi** [rzander/sccmclictr#156]
Could you run the Enable WinRM step automatically when connecting to a machine? It seems it is required for the connection to work, so it makes sense for it to attempt a connection, then if it fails, attempt to enable WinRM, then attempt the connection again, and only throw an error if all of those steps fail.

**Comment by horst789github** [comment]
yes, i need something like that. i have many servers where this is not activated or even after activating it does not work, but the access works via the very old client center. it would be very good if you could somehow implement a solution there.

**Comment by rzander** [comment]
WinRM is enabled on servers by default... If you have issues connecting devices over WinRM, talk with your it-admins... Maybe they have a GPO or CI in place to configure/disable WinRM. 
I think it's the wrong approach to use ClientCenter to configure WinRM on your systems...

**Comment by SCCMAdmin** [comment]
Chicken and the egg story people. You can't have CCCM enable WinRM for remote connections remotely. Microsoft would not allow it. It's a security issue/risk to allow things you can't connect in, to self-configure a connection and get in. 

You need to enable WinRM locally, then you can accept remote connections. There is heaps about it online and it appears to be a very misunderstood topic.

I agree with Roger here.

**Comment by mdoyle-cvi** [comment]
> Chicken and the egg story people. You can't have CCCM enable WinRM for remote connections remotely. Microsoft would not allow it. It's a security issue/risk to allow things you can't connect in, to self-configure a connection and get in.
> 
> You need to enable WinRM locally, then you can accept remote connections. There is heaps about it online and it appears to be a very misunderstood topic.
> 
> I agree with Roger here.

Except it has an "enable WinRM" button in the software, which enables WinRM on the remote PC. I'm just asking for an option in the program to enable it automatically when connecting, without having to click that button every time.

**Comment by SCCMAdmin** [comment]
I haven't seen it or ever used it. Where is it? Send a screen shot please. :)

**Comment by SCCMAdmin** [comment]
Oh yeah, found it. Honestly never ever used it. I will let you know how it goes in coming days.

You need to get WinRM configured in your entire environment in one hit, not machine by machine. That part of the answer doesn't really change.

**Comment by Talven81** [comment]
It works, sometimes... I use it on machines that for some reason haven't received/applied the proper GPOs.
